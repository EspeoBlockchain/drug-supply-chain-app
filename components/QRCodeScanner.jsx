import React, { PureComponent } from 'react';
import {
  StyleSheet,
  Text,
  View,
} from 'react-native';
import {
  BarCodeScanner,
  Permissions,
} from 'expo';
import PropTypes from 'prop-types';
import layout from '../constants/Layout';
import colors from '../constants/Colors';
import fonts from '../constants/Fonts';

const opacity = 'rgba(0, 0, 0, .6)';
const styles = StyleSheet.create({
  container: {
    flex: 0,
    flexBasis: layout.window.width + 65,
  },
  scannerContainer: {
    overflow: 'hidden',
    width: layout.window.width,
    height: layout.window.width,
  },
  scanner: {
    flex: 1,
  },
  title: {
    paddingVertical: 20,
    textAlign: 'center',
    backgroundColor: colors.transportLight,
    color: colors.white,
    fontFamily: fonts.main,
    alignSelf: 'flex-start',
    width: '100%',
    borderColor: colors.transportDark,
    borderWidth: 2,
    zIndex: 1,
    fontSize: 20,
    height: 65,
  },
  layerTop: {
    flex: 1,
    backgroundColor: opacity,
    width: '100%',
  },
  layerCenter: {
    flex: 1,
    flexDirection: 'row',
    width: '100%',
  },
  layerLeft: {
    flex: 1,
    backgroundColor: opacity,
  },
  focused: {
    flex: 2,
  },
  layerRight: {
    flex: 1,
    backgroundColor: opacity,
  },
  layerBottom: {
    flex: 1,
    backgroundColor: opacity,
    width: '100%',
  },
});

export default class QRCodeScreen extends PureComponent {
  state = {
    hasCameraPermission: null,
  }

  async componentDidMount() {
    const { status } = await Permissions.askAsync(Permissions.CAMERA);
    this.setState({ hasCameraPermission: status === 'granted' });
  }

  render() {
    const { hasCameraPermission } = this.state;
    const { title, handleQRScan } = this.props;

    if (hasCameraPermission === null) {
      return <Text style={styles.title}>Requesting for camera permission</Text>;
    }
    if (hasCameraPermission === false) {
      return <Text style={styles.title}>No access to camera</Text>;
    }
    return (
      <View style={styles.container}>
        <Text style={styles.title}>{title}</Text>
        <View style={styles.scannerContainer}>
          <BarCodeScanner
            onBarCodeScanned={handleQRScan}
            style={styles.scanner}
          >
            <View style={styles.layerTop} />
            <View style={styles.layerCenter}>
              <View style={styles.layerLeft} />
              <View style={styles.focused} />
              <View style={styles.layerRight} />
            </View>
            <View style={styles.layerBottom} />
          </BarCodeScanner>
        </View>
      </View>
    );
  }
}

QRCodeScreen.propTypes = {
  title: PropTypes.string,
  handleQRScan: PropTypes.func,
};

QRCodeScreen.defaultProps = {
  title: '',
  handleQRScan: () => null,
};
