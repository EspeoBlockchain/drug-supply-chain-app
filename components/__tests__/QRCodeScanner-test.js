import 'react-native';
import React from 'react';
import QRCodeScanner from '../QRCodeScanner';
import renderer from 'react-test-renderer';

it('renders corectly', () => {
    const tree = renderer.create(<QRCodeScanner />).toJSON();

    expect(tree).toMatchSnapshot();
});