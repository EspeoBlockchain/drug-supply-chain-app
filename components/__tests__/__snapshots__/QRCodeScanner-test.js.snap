// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders corectly 1`] = `
<Text
  style={
    Object {
      "alignSelf": "flex-start",
      "backgroundColor": "#cac6e5",
      "borderColor": "#9087d2",
      "borderWidth": 2,
      "color": "#fff",
      "fontFamily": "raleway",
      "fontSize": 20,
      "height": 65,
      "paddingVertical": 20,
      "textAlign": "center",
      "width": "100%",
      "zIndex": 1,
    }
  }
>
  Requesting for camera permission
</Text>
`;
